velox/exec/Task.cpp

```diff
@@ -1460,4 +1478,29 @@ ContinueFuture Task::requestPauseLocked(bool pause) {
   pauseRequested_ = pause;
   return makeFinishFutureLocked("Task::requestPause");
 }
+
+Task::TaskCompletionNotifier::~TaskCompletionNotifier() {
+  notify();
+}
+
+void Task::TaskCompletionNotifier::activate(
+    std::function<void()> callback,
+    std::vector<ContinuePromise> promises) {
+  active_ = true;
```

Comment:
Nit: Maybe a check that it is not already `active_` might help from an API standpoint.